addons/content-index/src/main/java/org/commonjava/indy/content/index/IndexingContentManagerDecorator.java

```diff
@@ -747,34 +751,8 @@ public Transfer store( final List<? extends ArtifactStore> stores, final StoreKe
                            final InputStream stream, final TransferOperation op, final EventMetadata eventMetadata )
             throws IndyWorkflowException
     {
-        Transfer transfer = delegate.store( stores, topKey, path, stream, op, eventMetadata );
-        if ( transfer != null )
-        {
-            if ( indexCfg.isEnabled() )
-            {
-                indexManager.indexTransferIn( transfer, topKey );
-            }
-
-            try
-            {
-                ArtifactStore topStore = storeDataManager.getArtifactStore( topKey );
-                nfc.clearMissing( new ConcreteResource( LocationUtils.toLocation( topStore ), path ) );
-
-                if ( indexCfg.isEnabled() )
-                {
-                    // We should deIndex the path for all parent groups because the new content of the path
-                    // may change the content index sequence based on the constituents sequence in parent groups
-                    indexManager.deIndexStorePath( topKey, path );
-                }
-            }
-            catch ( IndyDataException e )
-            {
-                Logger logger = LoggerFactory.getLogger( getClass() );
-                logger.error( String.format( "Failed to retrieve top store: %s for NFC management. Reason: %s",
-                                             topKey, e.getMessage()), e );
-            }
-        }
-
+        final Transfer transfer = delegate.store( stores, topKey, path, stream, op, eventMetadata );
+        deIndexQueue.put( System.currentTimeMillis(), new DeIndexInfo( transfer, null, topKey, TYPE_MULTI ) );
```

Comment:
I think we might need a UUID for the key, in case we have two store requests racing to completion at the same time. It would be rare, but that would also mean it might be really hard to detect the bug if they collided.

Reply:
Seems this queue solution not works well with bunch of lock timeout in ISPN. I think the async way of cache event in ISPN also has some blocking mechanism with the lock control, so I reverted this solution in this pr https://github.com/Commonjava/indy/pull/1475